src/main/java/de/metas/ui/web/handlingunits/process/WEBUI_M_HU_Pick.java

```diff
@@ -125,9 +137,87 @@ public Object getParameterDefaultValue(final IProcessDefaultParameter parameter)
 			lookupSource = LookupSource.lookup)
 	private LookupValuesList getShipmentScheduleValues(final LookupDataSourceContext context)
 	{
-		return createNewDefaultParametersFiller().getShipmentScheduleValues(context);
+
+		final HUEditorRow huEditorRow = getSingleHUEditorRow();
+
+		// Get all the matching shipment schedules and then filter for schedules which are not picked and match the order-line attributes.
+		return createNewDefaultParametersFiller()
+				.getShipmentScheduleValues(context)
+				.stream()
+				.filter(shipmentSchedule -> (!isShipmentSchedulePicked(shipmentSchedule.getIdAsInt())
+						&& doesHUAttribsMatchOrderLineAttribs(shipmentSchedule.getIdAsInt(), huEditorRow)))
+				.collect(LookupValuesList.collect());
+
+	}
+	
+	private HUEditorRow getSingleHUEditorRow() {
+		return streamSelectedRows()
+				.map(row -> HUEditorRow.cast(row))
+				.collect(GuavaCollectors.singleElementOrThrow(() -> new AdempiereException("only one selected row was expected")));
 	}
+	
+	// if picking candidate is available for the shipment schedule, then check if it is already picked
+		private boolean isShipmentSchedulePicked(int shipmentScheduleId)
```

Comment:
feel free to import our code formatter (and some other) rules: you can find them at https://github.com/metasfresh/metasfresh-dev/tree/master/eclipse-config

Reply:
Thanks @metas-ts. I've these configs setup in my eclipse. But somehow the formatting is messed during moving code from mykloudz repo to this branch. Will make sure going forward, such format issues shall not happen.